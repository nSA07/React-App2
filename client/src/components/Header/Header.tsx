import { FC } from "react";

export const Header: FC = () => {
  
  return (
    <header className="p-4 border-b">
      <h1 className="font-extrabold text-xl md:text-2xl">My Task Board</h1>
    </header>
  )
}
