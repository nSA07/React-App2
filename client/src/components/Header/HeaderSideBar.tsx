import { Button } from "@/components/ui/button"
import {
  Sheet,
  SheetContent,
  SheetTitle,
  SheetTrigger,
} from "@/components/ui/sheet"
import { ScrollArea } from "@/components/ui/scroll-area"

import { RotateCcw  } from 'lucide-react'
import { IHistory } from "@/types/types";
import { History } from "@/components/History/History";

export const HeaderSideBar = ({history}) => {  
           
    const data = [];
    function historyItem (item: IHistory) {
        item?.map(({ changes }) => {           
            changes?.map(({field, prev, next, taskName, dueData}) => { 

                data.push({field, prev, next, taskName, dueData})
            })
        });
    }
    
    historyItem(history);    

    return (
        <Sheet>
            <SheetTrigger asChild>
                <Button variant="outline" className="flex gap-2 font-medium">
                    <RotateCcw />
                    Історія
                </Button>
            </SheetTrigger>
            <SheetContent>
                <SheetTitle>Історія</SheetTitle>
                <ScrollArea className="w-full h-lvh rounded-md border p-4">
                    <ul className="p-2 flex flex-col gap-3">
                        {data?.map((item) => (
                            <History key={item.field} type={item.field} prev={item.prev} next={item.next} taskName={item.taskName} dueData={item.dueData} />
                        ))}
                    </ul>
                </ScrollArea>
            </SheetContent>
        </Sheet>
    )
}
