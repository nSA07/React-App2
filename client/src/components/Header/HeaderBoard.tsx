import { FC } from 'react'
import { HeaderSideBar } from "./HeaderSideBar";
import { useLocation } from "react-router-dom";
import { useGetHistoryQuery } from "@/redux"

export const HeaderBoard: FC = () => {
    const { state } = useLocation();
    const { data: history, error, isLoading } = useGetHistoryQuery(`history/?lists=${state.boardID}`)     

    return (
        <>
            {error ? (
                <>Oh no, there was an error</>
            ) : isLoading ? (
                <>Loading...</>
            ) : history ? (
                <HeaderSideBar history={history} />
            ) : null}
      </>
    )
}
