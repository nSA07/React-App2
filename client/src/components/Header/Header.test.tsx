import { render } from "@testing-library/react";
import { Header } from "./Header";

test('render <Header/> component', () => {
    const { container } = render(<Header />);
    expect(container.firstChild).toMatchSnapshot();
})