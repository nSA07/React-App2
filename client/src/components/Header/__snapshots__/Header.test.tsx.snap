// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Header/> component 1`] = `
<header
  class="p-4 border-b"
>
  <h1
    class="font-extrabold text-xl md:text-2xl"
  >
    My Task Board
  </h1>
</header>
`;
